services/net/filemonitor/FilemonitorAction.cs

```diff
@@ -88,6 +98,129 @@ public FileMonitorAction(IHttpRequestClient httpClient, IApiService api, IKafkaM
+    /// </summary>
+    /// <param name="manager"></param>
+
+    /// <returns></returns>
+    public async Task FetchFiles(IIngestServiceActionManager manager)
+    {
+        // Extract the ingest configuration settings
+        var filePattern = String.IsNullOrWhiteSpace(manager.Ingest.GetConfigurationValue("filePattern")) ? manager.Ingest.Source?.Code : manager.Ingest.GetConfigurationValue("filePattern");
+        filePattern = filePattern.Replace("<date>", $"{GetLocalDateTime(manager.Ingest, DateTime.Now.AddDays(manager.Ingest.GetConfigurationValue<double>("dateOffset"))):yyyyMMdd}");
+        Regex match = new Regex(filePattern);
+        if (String.IsNullOrWhiteSpace(filePattern)) throw new ConfigurationException($"Ingest '{manager.Ingest.Name}' is missing a 'filePattern'.");
+
+        // TODO: create new account to access server
+        // Extract the source connection configuration settings.
+        var remotePath = manager.Ingest.SourceConnection?.GetConfigurationValue("path");
+        var username = manager.Ingest.SourceConnection?.GetConfigurationValue("username");
+        var keyFileName = manager.Ingest.SourceConnection?.GetConfigurationValue("keyFileName");
+        var hostname = manager.Ingest.SourceConnection?.GetConfigurationValue("hostname");
+        if (String.IsNullOrWhiteSpace(hostname)) throw new ConfigurationException($"Ingest '{manager.Ingest.Name}' source connection is missing a 'hostname'.");
+        if (String.IsNullOrWhiteSpace(username)) throw new ConfigurationException($"Ingest '{manager.Ingest.Name}' source connection is missing a 'username'.");
+        if (String.IsNullOrWhiteSpace(keyFileName)) throw new ConfigurationException($"Ingest '{manager.Ingest.Name}' source connection is missing a 'keyFileName'.");
+        if (String.IsNullOrWhiteSpace(remotePath)) throw new ConfigurationException($"Ingest '{manager.Ingest.Name}' source connection is missing a 'path'.");
+
+        var sshKeyFile = Path.Combine(this.Options.PrivateKeysPath, keyFileName);
+
+        if (File.Exists(sshKeyFile))
+        {
+            var keyFile = new PrivateKeyFile(sshKeyFile);
+
+            var keyFiles = new[] { keyFile };
+            var connectionInfo = new ConnectionInfo(hostname,
+                                                    username,
+                                                    new PrivateKeyAuthenticationMethod(username, keyFiles));
+            using var client = new SftpClient(connectionInfo);
+            client.Connect();
+
+            var files = await FetchFileListing(client, remotePath);
+            files = files.Where(f => match.Match(f.Name).Success);
+
+            foreach (var file in files)
+            {
+                await CopyFile(client, manager.Ingest, Path.Combine(remotePath, file.Name));
+            }
+
+            client.Disconnect();
+        }
+        else
+        {
+            _logger.LogError("SSH Private key file does not exist: {file}", sshKeyFile);
```

Comment:
Should this throw an exception or is ignoring the issue and just logging suitable?

Reply:
Changed it to an exception, as the service can't do anything without the key file and should stop operating after max retries.